src/groupByKey.ts

```diff
@@ -2,6 +2,21 @@ interface Item {
   [key: string]: any;
 }
 
-export function groupByKey(items, field) {
-  // write code here;
+interface NewItem {
+  [key: string]: Item[];
+}
+
+export function groupByKey(items: Item[], field: keyof Item): NewItem {
+  const groupedObj: NewItem = {};
+
+  items.forEach((item) => {
+    if (!groupedObj[item[field]]) {
+      groupedObj[item[field]] = [];
+      groupedObj[item[field]].push(item);
+    } else {
+      groupedObj[item[field]].push(item);
+    }
```

Comment:
You can do all this things without `else`
```suggestion
    if (!groupedObj[item[field]]) {
      groupedObj[item[field]] = [];
    } 
      groupedObj[item[field]].push(item);
```

Reply:
Right, it looks much better!)